### DISCUSSION TITLE
Persistent fish abbreviation file

### DISCUSSION THREAD

**Discussion by kendofriendo** [fish-shell/fish-shell#9609] [Ideas]
Since universal abbreviations are no longer supported, could it be feasible to use a new config file loaded under ~/.config/fish/ e.g. abbreviations.fish where abbreviations are stored? This way, users can continue to use `$abbr foo` , and the abbreviation would be appended to the file.

I am currently utilizing a similar solution using a function wherein I echo the abbreviations into a file and source the file in config.fish, but it requires reloading the fish shell every time an abbreviation is added and manually editing the file if I need to remove one.

**Comment by ajgringo619** [comment]
It already exists. Create an abbreviations file in `$HOME/.config/fish/conf.d`. This will be read at shell start, can be sourced as needed. This will also keep your `$HOME/.config/fish/fish_variables` file cleaner.

**Reply by kendofriendo** [reply]
Thank you I will move the file there, but the other problems remain where I have to reload the environment  or source the file any time I add a new abbreviation, and I can't use the `abbr ` function or `abbr -e`.

**Comment by ajgringo619** [comment]
I guess you could write your own **abbr** function to add, then update your current shell's abbreviations. But this won't help on other open terminals.

**Comment by faho** [comment]
We do not want an automatically reloading file, sorry.

You *can* build something yourself, like

```fish
set -g _abbr_time (path mtime -- ~/.config/fish/conf.d/abbr.fish)
function reload_abbr --on-event fish_prompt
    set -l time (path mtime -- ~/.config/fish/conf.d/abbr.fish)
    or return
    if test "$time" -ne "$_abbr_time" 2>/dev/null
        abbr -e (abbr --list)
        source ~/.config/fish/conf.d/abbr.fish
        set -g _abbr_time $time
    end
end
```

which will reload your abbr.fish any time it changes on a new prompt.

But this isn't something we want to incorporate into fish proper as it will reexecute arbitrary code.
